backend/capellacollab/projects/toolmodels/modelsources/git/handler/handler.py

```diff
@@ -60,26 +62,65 @@ def get_artifact_from_job_as_content(
 
     @abc.abstractmethod
     async def get_file_from_repository(
-        self, project_id: str, trusted_file_path: str
+        self,
+        project_id: str,
+        trusted_file_path: str,
+        revision: str | None = None,
     ) -> bytes:
         pass
 
+    @abc.abstractmethod
+    def get_last_commit_id_for_file_path(
+        self, project_id: str, file_path: str, revision: str | None
+    ) -> t.Any:
+        pass
+
+    async def get_file_from_repository_or_artifacts_as_json(
+        self,
+        trusted_file_path: str,
+        job_name: str | None,
+        revision: str | None = None,
+    ) -> tuple[t.Any, dict]:
+        (
+            last_updated,
+            result,
+        ) = await self.get_file_from_repository_or_artifacts(
+            trusted_file_path, job_name, revision
+        )
+        return (last_updated, json.loads(result.decode("utf-8")))
+
     async def get_file_from_repository_or_artifacts(
-        self, trusted_file_path: str, job_name: str | None
-    ) -> bytes:
+        self,
+        trusted_file_path: str,
+        job_name: str | None,
```

Comment:
`job_name` shouldn't be optional here. The purpose of the function is to get the file from repository or artifacts. If there is no job_name, the user can call the get_file_from_repository function directly.